app/src/main/java/com/deitel/weatherviewer/MainActivity.java

```diff
@@ -44,6 +49,9 @@ protected void onCreate(Bundle savedInstanceState) {
       Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
       setSupportActionBar(toolbar);
 
+      mFusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
+
+
```

Comment:
Missing the permission check and the call to mFusedLocationClient.getLastLocation() ...